src/cli.rs

```diff
@@ -187,9 +187,21 @@ pub struct GitAddOpts {
 
 impl GitAddOpts {
     pub fn add(&self) -> Result<(String, Pin)> {
-        let url = Url::parse(&self.url)?;
+        let url = Url::parse(&self.url)
+            .map_err(|e| {
+                match e {
+                    url::ParseError::RelativeUrlWithoutBase => {
+                        anyhow::format_err!("URL scheme is missing. For git URLs, add the fully qualified scheme like git+ssh://. For local repositories, add file://")
+                    },
+                    url::ParseError::InvalidPort => {
+                        anyhow::format_err!("Invalid port number. For git URLs, try inserting a '/' after the ':' before the user name, like so: git+ssh://git@gitlab-instance.net:/user/repo.git")
+                    },
+                    e => e.into(),
+                }
+            })
+            .context("Failed to parse repository URL")?;
         let name = match url.path_segments().and_then(|x| x.rev().next()) {
-            None => return Err(anyhow::anyhow!("Path segment in URL missing.")),
+            None => anyhow::bail!("Path segment in URL missing."),
```

Comment:
Oh wow that one is cursed. Especially since *technically* this is a valid URL, except that it parses in an entirely different way than it should (`blitiri.xlumurb.eu` is the scheme).

What would be the "correct" version of that one? `git+ssh://blitiri.xlumurb.eu/~/git/wallpapers`?

Reply:
yes it works